Confine gnome-remote-desktop

bachradsusi opened the pull request:
- add new gnome_remote_desktop_t port mapping for tcp 3389-3399
- add file type for /var/lib/gnome-remote-desktop(/.*)?
- add new domain and transition for /usr/libexec/gnome-remote-desktop-daemon

zpytela commented on policy/modules/contrib/gnome_remote_desktop.if:
```diff
@@ -0,0 +1,155 @@
+	allow $1 gnome_remote_desktop_t:process { signal_perms };
+	ps_process_pattern($1, gnome_remote_desktop_t)
+
+    tunable_policy(`deny_ptrace',`',`
```
Can you use tabs as separators here as well?

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+	class tcp_socket { accept bind create getattr getopt listen name_bind setopt shutdown };
+	class unix_dgram_socket create;
+}
+
```
I believe this whole block is a local module legacy.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+# gnome_remote_desktop local policy
+#
+
+optional_policy(`
```
optional blocks should be after allow rules from base

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+
+files_read_etc_files(gnome_remote_desktop_t)
+
+miscfiles_read_localization(gnome_remote_desktop_t)
```
miscfiles is a module

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+
+#============= gnome_remote_desktop_t ==============
+allow gnome_remote_desktop_t gnome_remote_desktop_port_t:tcp_socket name_bind;
+allow gnome_remote_desktop_t self:tcp_socket { accept bind create getattr getopt listen read setopt write };
```
better use create_stream_socket_perms

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+allow gnome_remote_desktop_t gnome_remote_desktop_port_t:tcp_socket name_bind;
+allow gnome_remote_desktop_t self:tcp_socket { accept bind create getattr getopt listen read setopt write };
+allow gnome_remote_desktop_t self:unix_dgram_socket { create write };
+allow system_dbusd_t gnome_remote_desktop_t:tcp_socket { read write };
```
perhaps a room for using an interface? there are many other _rw_tcp_sockets()

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+fs_getattr_cgroup(gnome_remote_desktop_t)
+fs_getattr_xattr_fs(gnome_remote_desktop_t)
+init_read_state(gnome_remote_desktop_t)
+kerberos_read_config(gnome_remote_desktop_t)
```
kerberos is a module

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+kerberos_read_config(gnome_remote_desktop_t)
+kernel_dgram_send(gnome_remote_desktop_t)
+logging_write_syslog_pid_socket(gnome_remote_desktop_t)
+miscfiles_read_certs(gnome_remote_desktop_t)
```
one more miscfiles module

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+fs_getattr_xattr_fs(gnome_remote_desktop_t)
+init_read_state(gnome_remote_desktop_t)
+kerberos_read_config(gnome_remote_desktop_t)
+kernel_dgram_send(gnome_remote_desktop_t)
```
kernel does not obey to alphabetical order and is the first one

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+miscfiles_read_certs(gnome_remote_desktop_t)
+systemd_login_list_pid_dirs(gnome_remote_desktop_t)
+systemd_login_read_pid_files(gnome_remote_desktop_t)
+systemd_read_logind_sessions_files(gnome_remote_desktop_t)
```
systemd is a module

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+
+	dev_rw_dma_dev(xdm_t)
+	xserver_dbus_chat_xdm(gnome_remote_desktop_t)
+	xserver_dbus_chat_xdm(xdm_t)
```
is xdm_t to xdm_t dbus chat needed?

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,80 @@
+	xserver_dbus_chat_xdm(xdm_t)
+	xserver_read_xdm_state(gnome_remote_desktop_t)
+
+')
```
rules for xdm should rather go to xserver module, just one more interface will be needed

zpytela reviewed (changes requested):
Please apply the suggested changes or reply back.

bachradsusi commented:
@zpytela PTAL

I'll merge all the commits into one when you say it's ready to be merged.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,88 @@
+policy_module(gnome_remote_desktop, 1.0.0)
+
+gen_require(`
+	type gnome_remote_desktop_port_t;
```
This should not be needed. The build may have failed just because of #L38.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,88 @@
+files_read_etc_files(gnome_remote_desktop_t)
+
+#============= gnome_remote_desktop_t ==============
+allow gnome_remote_desktop_t gnome_remote_desktop_port_t:tcp_socket name_bind;
```
The port type is not defined in this module, so interface should be used.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,88 @@
+#============= gnome_remote_desktop_t ==============
+allow gnome_remote_desktop_t gnome_remote_desktop_port_t:tcp_socket name_bind;
+allow gnome_remote_desktop_t self:tcp_socket create_stream_socket_perms;
+allow gnome_remote_desktop_t self:unix_dgram_socket { create write };
```
Block of these allows should be at #L28.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,88 @@
+	gen_require(`
+		type system_dbusd_t;
+	')
+	allow system_dbusd_t gnome_remote_desktop_t:tcp_socket { read write };
```
Rule for system_dbus_t should be in the dbus module.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,88 @@
+		type xdm_t;
+		}
+
+	allow xdm_t gnome_remote_desktop_t:tcp_socket { getattr getopt read setopt shutdown write };
```
Rule for xdm_t should be in the xserver module.

zpytela commented on policy/users:
```diff
@@ -35,4 +35,7 @@ gen_user(sysadm_u, user, sysadm_r, s0, s0 - mls_systemhigh, mcs_allcats)
+gen_user(root, user, unconfined_r sysadm_r staff_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
+gen_user(unconfined_u, user, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
+gen_user(guest_u, user, guest_r, s0, s0)
+gen_user(xguest_u, user, xguest_r, s0, s0)
```
This has been merged in another PR.

zpytela reviewed (commented):
I am fine with squashing.

bachradsusi commented:
@zpytela PTAL

Please be more specific with other requirements. I'm kind of beginner in this and I need specific instructions.

zpytela commented on policy/modules/contrib/dbus.te:
```diff
@@ -410,3 +410,10 @@ allow session_bus_type dbusd_unconfined:dbus send_msg;
+	gen_require(`
+		type gnome_remote_desktop_t;
+	')
+	allow system_dbusd_t gnome_remote_desktop_t:tcp_socket { read write };
```
Please create a new interface and use it here, no declaration of types from outside this module would be needed then.

I haven't managed to find any good example elsewhere, so either we can use gnome_remote_desktop_rw_tcp_sockets() with rw_socket_perms and use it both for xdm_t and also here where only read and write are needed, or create another interface with an ugly name like gnome_remote_desktop_rwonly_tcp_sockets(). There are many _rw_tcp_sockets() interfaces, some of them use just "read write", some rw_socket_perms.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,66 @@
+
+files_read_etc_files(gnome_remote_desktop_t)
+
+kernel_dgram_send(gnome_remote_desktop_t)
```
Seems the 39e1762614 and 2495d43ebb crossed somehow or my hint was not clear enough: kernel should be before other interfaces from base.

zpytela commented on policy/modules/contrib/gnome_remote_desktop.te:
```diff
@@ -0,0 +1,66 @@
+fs_getattr_cgroup(gnome_remote_desktop_t)
+fs_getattr_xattr_fs(gnome_remote_desktop_t)
+init_read_state(gnome_remote_desktop_t)
+logging_write_syslog_pid_socket(gnome_remote_desktop_t)
```
Logging should be in an optional block.

zpytela commented on policy/modules/services/xserver.te:
```diff
@@ -1857,3 +1857,17 @@ tunable_policy(`selinuxuser_direct_dri_enabled',`
+
+	dev_rw_dma_dev(xdm_t)
+	xserver_dbus_chat_xdm(gnome_remote_desktop_t)
+	xserver_read_xdm_state(gnome_remote_desktop_t)
```
These two xserver_ calls should be in g_r_d.te in an optional block

bachradsusi commented:
@zpytela another round, PTAL

bachradsusi committed:
Confine gnome-remote-desktop

- add new gnome_remote_desktop_t port mapping for tcp 3389-3399
- add file type for /var/lib/gnome-remote-desktop(/.*)?
- add new domain and transition for /usr/libexec/gnome-remote-desktop-daemon

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2271661

zpytela commented:
Merging, thank you.

XhmikosR commented:
Thanks @bachradsusi and @zpytela for your work!

Do you plan to backport this patch to Fedora 40? I'm asking because it's supposed to fix https://bugzilla.redhat.com/show_bug.cgi?id=2271661 and it's a pity that without this patch GRD doesn't work OOTB on Fedora stable.

zpytela commented:
F40 backport is considered.
